File: packages/urdu-js/src/search/tokenize-search.ts
import { normalize } from '../normalization/normalize.js'

export function tokenizeForSearch(text: string): string[] {
  if (!text) return []
  return normalize(text)
    .split(/[\s\u060c\u061b]+/)
    .map(t => t.trim())
    .filter(t => t.length > 0)
}

File: packages/urdu-dart/lib/src/search/tokenize_search.dart
import '../normalization/normalize.dart';

/// Tokenizes text into an array of search-ready terms.
List<String> tokenizeForSearch(String text) {
  if (text.isEmpty) return [];
  return normalize(text)
      .split(RegExp(r'[\s\u060c\u061b]+'))
      .map((t) => t.trim())
      .where((t) => t.isNotEmpty)
      .toList();
}
